internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala

```diff
@@ -83,12 +105,11 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile with
         def processDependency(context: DependencyContext)(dep: ClassDependency): Unit = {
           val fromClassName = className(dep.from)
           def binaryDependency(file: File, onBinaryClassName: String) =
-            callback.binaryDependency(file, onBinaryClassName, fromClassName, sourceFile, context)
+            withCallback(_.binaryDependency(file, onBinaryClassName, fromClassName, sourceFile, context))
```

Comment:
It is not that simple sine AnalysisCallback is managed by zinc internally. 
@eed3si9n What in your opinion will best way to configure thread-handling there?